lib/plugins/aws/lib/monitorStack.js

```diff
@@ -25,127 +24,131 @@ module.exports = {
 
     this.serverless.cli.log(`Checking Stack ${action} progress...`);
 
-    return new BbPromise((resolve, reject) => {
-      async.whilst(
-        () => validStatuses.indexOf(stackStatus) === -1,
-        callback => {
-          const cfStackStatusCheckFrequency =
-            options.frequency || process.env.SLS_AWS_MONITORING_FREQUENCY || 5000;
+    return new Promise((resolve, reject) => {
```

Comment:
Let's remove this promise wrapper (note `recursivePromiseFn()` already returns promise, and we should just return it's result)